backend/open_webui/routers/retrieval.py

```
@@ -1529,3 +1846,1159 @@ def process_files_batch(
+        kb_referenced_files = set()
+        try:
+            for kb in existing_knowledge_bases:
+                if kb.data and isinstance(kb.data, dict):
+                    file_ids = kb.data.get("file_ids", [])
+                    if isinstance(file_ids, list):
+                        kb_referenced_files.update(file_ids)
+
+            log.info(
+                f"Found {len(kb_referenced_files)} files referenced in knowledge bases (will be preserved)"
+            )
+        except Exception as e:
```

Ruff (0.12.2)

Do not catch blind exception: `Exception`

(BLE001)